### PR TITLE
Add recipe for go compiler activation

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for review, I'm removing the activation in the go-feedstock here: https://github.com/conda-forge/go-feedstock/pull/116 This all enables cross-compilation for go-recipes following along how we did build it for rust recipes.

Azure has passed: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366379&view=results

File: recipes/go-activation/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{% set cross_target_platform = "foo" %}
+{% endif %}
+
+package:
+  name: go-{{ go_variant_str }}_{{ cross_target_platform }}
+  version: {{ version }}
+
+build:
+  number: 1
+  skip: True  # [linux and ((cross_target_platform == "osx-arm64") or (cross_target_platform == "osx-64")) and cgo]
```
Comment (isuruf) [inline recipes/go-activation/meta.yaml:15]: Why this skip?

File: recipes/go-activation/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (isuruf) [inline recipes/go-activation/meta.yaml:16]: Can you add a `run_exports: strong_constrained: __osx>=10.12`?

File: recipes/go-activation/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+cross_target_platform:
+  - linux-64                            # [linux]
+  - linux-aarch64                       # [linux]
+  - linux-ppc64le                       # [linux]
+  - osx-64                              # [linux or (osx and x86_64)]
+  - osx-arm64                           # [linux or (osx and x86_64) or (osx and arm64)]
+  - win-64                              # [win]
```
Comment (isuruf) [inline recipes/go-activation/conda_build_config.yaml:7]: Following doesn't work?
```suggestion
  - win-64                              # [linux64 and win]
```

File: recipes/go-activation/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipes/go-activation/meta.yaml:15]: I thought that we cannot compile C code from Linux to OSX but actually, we can, so we can remove this skip.
```suggestion
```

File: recipes/go-activation/conda_build_config.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+cross_target_platform:
+  - linux-64                            # [linux]
+  - linux-aarch64                       # [linux]
+  - linux-ppc64le                       # [linux]
+  - osx-64                              # [linux or (osx and x86_64)]
```
Comment (isuruf) [inline recipes/go-activation/conda_build_config.yaml:5]: Following should work right?
```suggestion
  - osx-64                              # [linux or (osx and x86_64) or (osx and arm64)]
```

**Comment by isuruf** [comment]
Is it possible to add a simple hello world compile to see if cross-compiling works?

**Comment by xhochy** [comment]
CI has passed on all platforms: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366544&view=results

File: recipes/go-activation/run_test.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+# the existence of SRC-DIR for an old behaviour change.
+test "$(go env GOPATH)" == "${PWD}/gopath"
+
+
+# Print diagnostics
+go env
+
+go mod init example.com/hello_world
+go build .
+if [[ "${cross_target_platform}" == "${build_platform}" ]]; then
```
Comment (isuruf) [inline recipes/go-activation/run_test.sh:23]:
```suggestion
if [[ "${cross_target_platform}" == "${build_platform}" || "${CMAKE_CROSSCOMPILING_EMULATOR}" != "" ]]; then
```

**Review by isuruf — COMMENTED** [review]
Looks good.

Let's wait for,
1. go-feedstock PR to merge
2. Need a PR to add `go-activation` to https://github.com/conda-forge/feedstock-outputs/blob/master/outputs/g/o/c/go-cgo_linux-aarch64.json and similar files.

**Comment by xhochy** [comment]
Output PR: https://github.com/conda-forge/feedstock-outputs/pull/22

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module example.com/hello
+
+go 1.17
```
Comment (jakirkham) [inline go.mod:3]: Was this meant to be in the recipe or did we want this in staged-recipes for some reason?

Comment (xhochy) [reply go.mod:3]: Ah, this was an accidental commit. This will get autogenerated during the tests. I'll revert it on the feedstock.

Comment (jakirkham) [reply go.mod:3]: All good 🙂 Submitted PR ( https://github.com/conda-forge/staged-recipes/pull/15978 ) to drop this from staged-recipes

**Pull request merged**